### DISCUSSION TITLE
IndexError in the best_solution function in line 3124

### DISCUSSION THREAD

**Discussion by augustodamasceno** [ahmedfgad/GeneticAlgorithmPython#93] [Q&A] [answered]
After some time running the GA function ga_instance.run(), a IndexError occur in the following line:

`best_match_idx = numpy.where(pop_fitness == numpy.max(pop_fitness))[0][0]`

Has anyone come across this index error before? How did you resolve it?

After I modified the fitness parameters to a more appropriate setting I was able to get GA to run for longer, however the Index error still persists. 

Thanks for anyone's help.

**Comment by ahmedfgad** [comment]
Hi @augustodamasceno,

It it possible to share the code?

**Reply by augustodamasceno** [reply]
Thanks. Problem fixed change keep_parents from 1 to -1.

**Comment by augustodamasceno** [comment, marked as answer]
Problem fixed changing keep_parents from 1 to -1.

I think the problem was the population size decreasing to 0.
